### ISSUE TITLE
AQL compilation error - exception during macro expansion

### ISSUE THREAD

**Issue by pawank** [outr/scarango#16]
Hello,

Settings used:
Scala: 2.12.10
Play framework: 2.7
Scarango lib: "com.outr" %% "scarango-driver" % "2.1.0"

The following error was thrown in compiling below code.
```
[error] /home/pk/app/models/daos/AuthTokenDAOImpl.scala:61:20: exception during macro expansion:
[error] scala.MatchError: play.assets { (of class java.lang.String)
[error]         at profig.ProfigLookupPath$.$anonfun$propertiesString2Json$2(ProfigLookupPath.scala:108)
[error]         at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[error]         at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[error]         at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[error]         at profig.ProfigLookupPath$.propertiesString2Json(ProfigLookupPath.scala:99)
[error]         at profig.ProfigLookupPath$.toJson(ProfigLookupPath.scala:187)
[error]         at profig.ProfigLookupPath$.$anonfun$toJsonStrings$1(ProfigLookupPath.scala:60)
[error]         at scala.collection.immutable.List.map(List.scala:290)
[error]         at profig.ProfigLookupPath$.toJsonStrings(ProfigLookupPath.scala:55)
[error]         at com.outr.arango.AQLMacros$.aql(AQLMacros.scala:16)
[error]     val queryAQL = aql"FOR p IN $collectionName FILTER p.id == $pk RETURN p"
[error]                    ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 2 s, completed 22 Nov, 2019 2:58:25 PM
```
Source code: 

```
/**
   * Finds a token by its ID.
   *
   * @param id The unique token ID.
   * @return The found token or None if no token for the given ID could be found.
   */
  def find(id: UUID): Future[Option[AuthToken]] = {
    import com.outr.arango._
    import com.outr.arango.api.APITransaction
    val collectionName: String = "tokens"
    val pk = id.toString()
    val queryAQL = aql"FOR p IN $collectionName FILTER p.id == $pk RETURN p"
    Future.successful(None)
  }
```  

AQL without any args expansion too is throwing the same error.

Any help is appreciated.

**Comment by darkfrog26** [comment]
It looks like you have a `.properties` file that `Profig` is trying to load at compile-time that has `play.assets` in it, and it can't parse it.  The file would likely be `config.properties`, `configuration.properties`, `app.properties`, `application.properties`, or `defaults.properties`.  These are auto-loaded by `Profig`.

**Comment by darkfrog26** [comment]
If you can rename the properties file with another name that `Profig` won't pick up, or remove that un-comprehendible block, or send me the block and I can update `Profig` to better support that scenario.  Any of those three options should resolve your issue.

**Comment by pawank** [comment]
There is only 1 config file `application.conf` which is play framework default one and also not used in `Profig`.  
Sharing the git application to reproduce the error: 
https://github.com/pawank/scala-play-silhouette-arangodb
How to run the project:
`sbt run`  

There are 2 errors:
1)
```
[error] /Users/pawan/git/github/pawank/scala-play-silhouette-arangodb/app/models/daos/AuthTokenDAOImpl.scala:67:20: exception during macro expansion:
[error] scala.MatchError: play.assets { (of class java.lang.String)
[error]         at profig.ProfigLookupPath$.$anonfun$propertiesString2Json$2(ProfigLookupPath.scala:108)
[error]         at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[error]         at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[error]         at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[error]         at profig.ProfigLookupPath$.propertiesString2Json(ProfigLookupPath.scala:99)
[error]         at profig.ProfigLookupPath$.toJson(ProfigLookupPath.scala:187)
[error]         at profig.ProfigLookupPath$.$anonfun$toJsonStrings$1(ProfigLookupPath.scala:60)
[error]         at scala.collection.immutable.List.map(List.scala:290)
[error]         at profig.ProfigLookupPath$.toJsonStrings(ProfigLookupPath.scala:55)
[error]         at com.outr.arango.AQLMacros$.aql(AQLMacros.scala:16)
[error]     val queryAQL = aql"FOR p IN $collectionName FILTER p.id == $pk RETURN p"
[error]                    ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```  
because of AQL usage which internally using Profig and not able to parse some config file(s) (as mentioned by you)

2) 
Another error is thrown if I comment the line causing the error `val queryAQL = aql"FOR p IN $collectionName FILTER p.id == $pk RETURN p"` from file, `scala-play-silhouette-arangodb/app/models/daos/AuthTokenDAOImpl.scala` . 
```
1) Error injecting constructor, java.lang.RuntimeException: Failed to decode from {

}
  at models.repos.ArangoDbService.<init>(ArangoDbService.scala:61)
  at modules.SilhouetteModule.configure(SilhouetteModule.scala:124) (via modules: com.google.inject.util.Modules$OverrideModule -> modules.SilhouetteModule)
  while locating models.repos.ArangoDbService

1 error]
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:214)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:144)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:304)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:189)
        at akka.stream.impl.fusing.MapAsync$$anon$30.onPush(Ops.scala:1261)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:523)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:409)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:606)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:485)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:581)
Caused by: com.google.inject.CreationException: Unable to create injector, see the following errors:

1) Error injecting constructor, java.lang.RuntimeException: Failed to decode from {

}
```
Again, it looks like config file parsing issue maybe I am wrong on it.
If I replace Scarango lib and use Arangodb java driver lib then all works fine.  

Thanks

**Comment by darkfrog26** [comment]
The first issue I understand and should be able to fix this week. However, the second issue never references any of my libs, so I looked at your code, and I'm guessing you've changed it since you produced this error: https://github.com/pawank/scala-play-silhouette-arangodb/blob/5a311e0c7a8df16134ffa14dfe281fb5f9f1a890/app/models/repos/ArangoDbService.scala#L61

**Comment by darkfrog26** [comment]
@pawank, this issue should be resolved if you upgrade to `2.3.0`.  Please let me know if you continue to have issues.
